rfcs/xxxx-content-type-definition.md

```diff
@@ -0,0 +1,112 @@
+- Start Date: 2021-05-21
+- RFC PR: (leave this empty)
+
+# Summary
+
+## Summary
+
+We will introduce an improved version of the content type definition files. In this RFC we are only describing the attributes types and not the options for now. We hope to make them more understandable and usable both manually and programmaticaly (CLI / parsing ...)
+
+## Example
+
+```js
+const articleDef = {
+  attributes: {
+    title: {
+      type: 'string',
+    },
+    seo: {
+      type: 'component',
+      component: 'seo',
+    },
+    body: {
+      type: 'dynamiczone',
+      components: ['txt.image-text', 'txt.rich'],
+    },
+    images: {
+      type: 'media',
+      multiple: true,
+    }
+    category: {
+      type: 'relation',
+      relation: 'manyToOne',
+      target: 'category',
+    },
+  },
+};
+```
+
+## Motivation
+
+The main goal of this change is to simplify readability of the content-type files so developers understand better what they are working with. This will also help make the internal database layer more standard for future evolution.
+
+We will also start reserving some property names for later features (e.g `parsing`, `formating`, `validation`)
+
+## Detailed design
+
+Every attribute **MUST** have a `type` property.
+
+### Medias
+
+To simplify the usage of media attributes we are also simplifying the corresponding content-type attribute.
+
+```js
+const articleDef = {
+  attributes: {
+    images: {
+      type: "media",
+      multiple: true | false,
+    },
+  },
+};
+```
+
+### Relations
+
+The major change is for relations that will be change to use a `type` property of relation and a `relation` property to describe which relation it is.
+
+**Examples**
+
+**Many `Articles` have One `Category`**
+
+```js
+const articleDef = {
+  attributes: {
+    category: {
+      type: "relation",
+      relation: "manyToOne",
+      target: "category",
+    },
+  },
+};
+```
+
+This relation is in one direction only (from Article to Category). It means you cannot fetch or edit the `articles` from a `category` in the Admin Panel.
+
+To make the relation visible in both directions you can use the following setup
+
+```js
+const articleDef = {
+  attributes: {
+    category: {
+      type: "relation",
+      relation: "manyToOne",
+      target: "category",
+      inversedBy: "articles",
+    },
+  },
+};
+
+const categoryDef = {
+  attributes: {
+    category: {
+      type: "relation",
+      relation: "oneToMany",
+      target: "article",
+      mappedBy: "category",
```

Comment:
I am not sure to understand the `mappedBy` property, can we have a definition of relation that only uses `mappedBy` on one side?

```js
const articleDef = {
  attributes: {
    category: { type: 'relation', relation: 'oneToMany', target: 'category' }
  }
}

const catDef = {
  attributes: { type: 'relation', relation: 'oneToMany', targert: 'article', mappedBy: 'category' } 
}
```
  }
```

Reply:
I'm also with you at @soupette; @alexandrebodin where did the mappedBy name come from? I'm wondering if something like `targetField` would be a bit more clear?

Reply:
the inversedBy / mappedBy  keys come as a pair. to know which side is owner and make sure tje inverse side is pointing to the valid field name.

This naming comes from several ORMs like doctrine / mikro-orm. They are pretty common in the ORM ecosystem and easy to define. We could find something clearer. We are open to suggestions. 
- `targetField` is incorrect as we don't target a field on the other entity but the entity itself. the field on the other side is just a way to access the relation for the other side.